soccer/planning/CompositePath.cpp

```diff
@@ -0,0 +1,147 @@
+#include "CompositePath.hpp"
+
+using namespace std;
+namespace Planning
+{
+	CompositePath::CompositePath(unique_ptr<Path> path) {
+		append(std::move(path));
+	}
+
+	CompositePath::CompositePath(Path *path) {
+		append(path);
+	}
+
+	void CompositePath::append(Path *path) {
+		append(std::unique_ptr<Path>(path));
+	}
+
+	void CompositePath::append(unique_ptr<Path> path) {
+		if (path && path->getTime()>0){
+			paths.push_back(std::move(path));
+		}
+	}
+	/**
+	 * A path describes the position and velocity a robot should be at for a
+	 * particular time interval.  This method evalates the path at a given time and
+	 * returns the target position and velocity of the robot.
+	 *
+	 * @param[in] 	t Time (in seconds) since the robot started the path
+	 * @param[out] 	targetPosOut The position the robot would ideally be at at the given time
+	 * @param[out] 	targetVelOut The target velocity of the robot at the given time
+	 * @return 		true if the path is valid at time @t, false if you've gone past the end
+	 */
+	bool CompositePath::evaluate(float t, Geometry2d::Point &targetPosOut, Geometry2d::Point &targetVelOut) const 
+	{
+		if (paths.empty()) {
+			return false;
+		}
+		if (t<0) {
+			return false;
+		}
+		for (const std::unique_ptr<Path> &path: paths)
+		{
+			float timeLength = path->getTime();
+			t -= timeLength;
+			if (t<=0 || timeLength == -1) {
+				t += timeLength;
+				path->evaluate(t, targetPosOut, targetVelOut);
+				return true;
+			}
+		}
+		return false;
+	}
+
+	/**
```

Comment:
It's probably best not to copy the doxygen comments from the header file into the .cpp file b/c they'll inevitably get out of sync as changes are made.